src/compas_timber/elements/beam.py

```diff
@@ -648,3 +648,51 @@ def get_dimensions_relative_to_side(self, ref_side_index):
         if ref_side_index in [1, 3]:
             return self.height, self.width
         return self.width, self.height
+
+    def front_side(self, ref_side_index):
+        """Returns the front side of the beam relative to the given reference side.
+        This method does not consider the start and end sides.
+
+        Parameters
+        ----------
+        ref_side_index : int
+            The index of the reference side to which the front side should be calculated.
+
+        Returns
+        -------
+        frame : :class:`~compas.geometry.Frame`
+            The frame of the front side of the beam relative to the reference side.
+        """
+        return self.ref_sides[(ref_side_index + 1) % 4]
+
+    def back_side(self, ref_side_index):
+        """Returns the back side of the beam relative to the given reference side.
+        This method does not consider the start and end sides.
+
+        Parameters
+        ----------
+        ref_side_index : int
+            The index of the reference side to which the back side should be calculated.
+
+        Returns
+        -------
+        frame : :class:`~compas.geometry.Frame`
+            The frame of the back side of the beam relative to the reference side.
+        """
+        return self.ref_sides[(ref_side_index - 1) % 4]
+
+    def opp_side(self, ref_side_index):
+        """Returns the opposite side of the beam relative to the given reference side.
```

Comment:
See the class level dictionary `OPPOSITE_MAP`, if it's not needed then just remove it.